app/Jobs/MovePipeline/UnfollowLegacyAccountMovePipeline.php

```diff
@@ -0,0 +1,157 @@
+            ->whereNull('profiles.deleted_at')
+            ->select('profiles.id', 'profiles.user_id', 'profiles.username', 'profiles.private_key', 'profiles.status')
+            ->chunkById(100, function ($followers) use ($actor, $addlHeaders, $targetInbox, $targetPid) {
+                $client = new Client([
+                    'timeout' => config('federation.activitypub.delivery.timeout'),
+                ]);
+                $requests = function ($followers) use ($client, $actor, $addlHeaders, $targetInbox, $targetPid) {
+                    $activity = [
+                        '@context' => 'https://www.w3.org/ns/activitystreams',
+                        'type' => 'Undo',
+                        'id' => null,
+                        'actor' => null,
+                        'object' => [
+                            'type' => 'Follow',
+                            'id' => null,
+                            'object' => $actor,
+                            'actor' => null,
+                        ],
+                    ];
+                    foreach ($followers as $follower) {
+                        if (! $follower->private_key || ! $follower->username || ! $follower->user_id || $follower->status === 'delete') {
+                            continue;
+                        }
+                        $permalink = 'https://'.config('pixelfed.domain.app').'/users/'.$follower->username;
+                        $activity['id'] = $permalink.'#follow/'.$targetPid.'/undo';
+                        $activity['actor'] = $permalink;
+                        $activity['object']['id'] = $permalink.'#follows/'.$targetPid;
+                        $activity['object']['actor'] = $permalink;
+                        $keyId = $permalink.'#main-key';
+                        $payload = json_encode($activity);
+                        $url = $targetInbox;
+                        $headers = HttpSignature::signRaw($follower->private_key, $keyId, $targetInbox, $activity, $addlHeaders);
+                        yield function () use ($client, $url, $headers, $payload) {
+                            return $client->postAsync($url, [
+                                'curl' => [
+                                    CURLOPT_HTTPHEADER => $headers,
+                                    CURLOPT_POSTFIELDS => $payload,
+                                    CURLOPT_HEADER => true,
+                                ],
+                            ]);
+                        };
+                    }
+                };
+
+                $pool = new Pool($client, $requests($followers), [
+                    'concurrency' => config('federation.activitypub.delivery.concurrency'),
+                    'fulfilled' => function ($response, $index) {},
+                    'rejected' => function ($reason, $index) {},
+                ]);
+
+                $promise = $pool->promise();
+
+                $promise->wait();
+            }, 'id');
```

Comment:
@dansup I know you're trying to get this feature out the door, and I'm new to this project (not to Laravel), but I want to ask constructively:
1. Would this logic be better in its own queued Job and the loop body a quick `dispatch(WellNamedJob::class);`
2. If any of the Guzzle requests fail, would it affect the chunking query as a whole? i.e would the rest of the move logic fail on the first follower failure? Would that be desireable?
3. I mention the separate job because they can be individually restarted via Horizon if they fail and that seems like it could be nice here, to my Laravel-eye.

Apologies, if this is already on your burndown list / radar.

Reply:
@cmack Yeah that is a great idea, am finishing up the refactor as we speak!

Reply:
Shipped in 15918cba!